state/executor.go

```diff
@@ -370,10 +365,6 @@ func (t *Transition) apply(msg *types.Transaction) (
 	if err != nil {
 		return nil, 0, false, err
 	}
-	// TODO: Check if this is even possible
-	if leftoverGas > msg.Gas {
-		return nil, 0, false, errorVMOutOfGas
-	}
```

Comment:
I'm afraid we cannot do this, it is the only check that we have if intrinsic gas cost goes above `msg.Gas`.
If you take a look at `preCheck()` method on line 349 it deducts the `msg.Gas` from `intrinsicGas` cost of the transaction it calculated.

If the intrinsic gas is greater than the gas limit that sender has provided through `msg.Gas`, then `gasAvailable` will go through underflow, and end up being some big number.
That's why we still need this weird check on the outside to check if the `leftoverGas` is greater than `msg.Gas` because then we know that what has occurred is that transaction has run out of gas during intrinsic gas cost check.

Please leave this as it is, I have a fix for this as a part of my task 🙂

Reply:
What saddens me is that our tests still pass, even with this removed. 🙃